src/app/ps/persistent_storage_json.hpp

```diff
@@ -0,0 +1,314 @@
+/*
+ *    Copyright (c) 2020, The OpenThread Commissioner Authors.
+ *    All rights reserved.
+ *
+ *    Redistribution and use in source and binary forms, with or without
+ *    modification, are permitted provided that the following conditions are met:
+ *    1. Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *    2. Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *    3. Neither the name of the copyright holder nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   The file defines json-based persistent storage.
+ */
+
+#ifndef _PERSISTENT_STORAGE_JSON_HPP_
+#define _PERSISTENT_STORAGE_JSON_HPP_
+
+#include "persistent_storage.hpp"
+#include "semaphore.hpp"
+
+#include "nlohmann_json.hpp"
+
+#include <algorithm>
+#include <functional>
+
+namespace ot {
+
+namespace commissioner {
+
+namespace persistent_storage {
+
+/**
+ * Implements PersistentStorage interface based on single json file
+ *
+ * Default persistent storage implemetaion used by Registry
+ * @see registry.hpp
+ * @see registry_entries.hpp
+ * @see persistent_storage.hpp
+ */
+class PersistentStorageJson : public PersistentStorage
+{
+public:
+    /**
+     * Constructor
+     *
+     * @param[in] aFileName file name to be used as storage
+     */
+    PersistentStorageJson(std::string const &aFileName);
+
+    /**
+     * Destructor
+     */
+    virtual ~PersistentStorageJson();
+
+    /**
+     * Opens file, loads cache
+     *
+     * @return Status
+     * @see Status
+     */
+    virtual Status Open() override;
```

Comment:
```suggestion
   Status Open() override;
```

Use only `override` in derived classes.

Reply:
Fixed